calcom/cal.com
# fix: scheduleId not unset properly bug

## What does this PR do?

* Deprecates useHostSchedulesForTeamEvent from metadata
* Fixes issue with scheduleId remaining after useHostSchedulesForTeamEvent toggled.

How to test:
* Use current main and attempt to get scheduleId set and desynced (simply enable the schedule, then set it, then untoggle).
* Checkout the fix, check behaviour (It should now show the schedule was in fact still enabled)

Fixup old affected event types:
https://github.com/calcom/cal.com/pull/15993/commits/81ce59deb1f2206fc5e88e6b02a2bd9bd5e9fee2

This is not done as a migration as migrations should be able to be ran at any time, however when running this migration after this fix this may unset schedules of event types that are working correctly.
